### ISSUE TITLE
Webview.maui.blazorwebview.blazorwebviewinitializing ,WebView Runtime , BrowserExecutableFolder

### ISSUE THREAD

**Issue by carsten-riedel** [dotnet/maui#15304]
### Description

On windows desktop platform blazorwebview.blazorwebviewinitializing has no effect.

As stated on the official documentation. https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.webview.maui.blazorwebview.blazorwebviewinitializing?view=net-maui-7.0 

`Raised before the web view is initialized. On some platforms this enables customizing the web view configuration. `

- Please list current platforms where this has effect.

- Please make this feature available for MAUI windows desktop applications.

### Public API Changes

Enable Webview.maui.blazorwebview.blazorwebviewinitializing for windows desktop

### Intended Use-Case

I had recently some problems with the Datalist from WebView Runtime inside MAUI app. 

See for reference here. https://github.com/MicrosoftEdge/WebView2Feedback/issues/3496 

I wanted to influence the WebView Runtime Startup parameters. To use another **BrowserExecutableFolder**

See https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.webview.blazorwebviewinitializingeventargs?view=net-maui-7.0 

In this case i followed the MAUI documentations first, but that only led me to a dead end.

I need some relay reliable **official way**. To influence the WebView Runtime versions used, if I rollout an application to production and then there is an issue with the underlying WebView runtime there is no official documentation how to deal with it. The blazorwebview.blazorwebviewinitializing seemed to be supposed for that but yeah it basically not works for windows desktop.

**Comment by Eilon** [comment]
I noticed the same thing recently while trying to use WebView2 Fixed Runtime. I was certain I tried using BlazorWebViewInitializing at one point and it worked as expected. So either I mis-remembered it, or maybe something changed in BlazorWebView (somewhat unlikely that this specific area changed), or in WebView2 (certainly possible).

So for now the best workaround I'm aware of (that you've seen as well) is to set the environment variable (which affects all WebView2's in the app, which is likely what you want anyway):

```c#
        Environment.SetEnvironmentVariable("WEBVIEW2_BROWSER_EXECUTABLE_FOLDER", @"C:\Temp\WebViewRuntime\Microsoft.WebView2.FixedVersionRuntime.113.0.1774.57.x64");
```

**Comment by richardhauer** [comment]
This is still an issue, with net10 and v10.0.71 of the `Microsoft.AspNetCore.Components.WebView.Maui` library.

The problem is here:
https://github.com/dotnet/maui/blob/1c0c0eb20018ca9f37b01cc8125998134094bab6/src/BlazorWebView/src/SharedSource/WebView2WebViewManager.cs#L264

The `WebView2WebViewManager::TryInitializeWebView2()` method has 2 forks:
- `#if WEBVIEW2_MAUI`
- `#elif WEBVIEW2_WINFORMS || WEBVIEW2_WPF`

In both cases the webview is initialised with `await _webview.EnsureCoreWebView2Async()`, but the environment options that are populated in the `BlazorWebViewInitializing` event is only passed to the `Ensure` method in the WinForms and WPF branch via `await _webview.EnsureCoreWebView2Async(_coreWebView2Environment)`. In the MAUI branch the method overload with no arguments are used so the browser process is initialised without the settings.

I don't see any possible workaround in the code under MAUI. It looks pretty intentional to me, so there's probably a reason this has been done this way.

The only other approach might be to set process-level environment variables to coerce the browser process to pick up the flags.

**Comment by carsten-riedel** [comment]
Solution was eighter envars or additional startup parameters as far as i can remember.

**Comment by richardhauer** [comment]
@carsten-riedel Yes I was able to get over the line by adding an environment variable in ‘initializing’ and removing it again in ‘initialized’.

I wouldn’t call that a “solution” though - more of a kludge. It’s clearly not the intended approach.

**Comment by carsten-riedel** [comment]
> [@carsten-riedel](https://github.com/carsten-riedel) Yes I was able to get over the line by adding an environment variable in ‘initializing’ and removing it again in ‘initialized’.
> 
> I wouldn’t call that a “solution” though - more of a kludge. It’s clearly not the intended approach.

sure but my inital request is from 2023, did i like it nope, did anybody change it nope, should it be too difficult to change nope

**Comment by richardhauer** [comment]
Sadly that’s about par for the course. I like to leave my findings in GH in the hopes someone takes an interest, or at least it helps someone out.

**Comment by carsten-riedel** [comment]
> Sadly that’s about par for the course. I like to leave my findings in GH in the hopes someone takes an interest, or at least it helps someone out.

Yeah, I think you’re right that fixing the version properly is the best approach. I ran into a few WebView-specific quirks, so for now manually selecting the shipped versions seems to be the only reliable workaround.
